File: src/main/java/fi/aalto/cs/apluscourses/model/NewsTree.java
package fi.aalto.cs.apluscourses.model;

import java.util.Collections;
import java.util.List;
import org.jetbrains.annotations.NotNull;

public class NewsTree {
  @NotNull
  private final List<News> news;

  public NewsTree() {
    this(Collections.emptyList());
  }

  public NewsTree(@NotNull List<News> news) {
    this.news = news;
  }

  @NotNull
  public List<News> getNews() {
    return news;
  }

  public void setAllRead() {
    news.forEach(News::setRead);
  }

  public long unreadCount() {
    return news.stream().filter(n -> !n.isRead()).count();
  }

}

File: src/main/kotlin/fi/aalto/cs/apluscourses/model/news/NewsTree.kt
package fi.aalto.cs.apluscourses.model.news

class NewsTree(val news: List<NewsItem> = emptyList()) {
    fun setAllRead() {
        news.forEach { newsItem: NewsItem -> newsItem.isRead = true }
    }

    fun setRead(id: Long) {
        news.find { n: NewsItem -> n.id == id }?.isRead = true
    }

    fun unreadCount(): Int {
        return news.count { n: NewsItem -> !n.isRead }
    }
}
